test/v2/mocks/MockHttpClient.scala

```diff
@@ -41,6 +42,40 @@ trait MockHttpClient extends MockFactory {
         }})
     }
 
+    def parameterGet[T](url: String,
+                        parameters: Seq[(String, String)],
+                        config: HeaderCarrier.Config,
+                        requiredHeaders: Seq[(String, String)] = Seq.empty,
+                        excludedHeaders: Seq[(String, String)] = Seq.empty): CallHandler[Future[T]] = {
+      (mockHttpClient
+        .GET(_: String, _: Seq[(String, String)], _: Seq[(String, String)])(_: HttpReads[T], _: HeaderCarrier, _: ExecutionContext))
+        .expects(where {
+          (actualUrl: String, params: Seq[(String, String)], _: Seq[(String, String)], _: HttpReads[T], hc: HeaderCarrier, _: ExecutionContext) => {
+            val headersForUrl = hc.headersForUrl(config)(actualUrl)
+            url == actualUrl &&
+              requiredHeaders.forall(h => headersForUrl.contains(h)) &&
```

Comment:
Since you're updating this class, can you instead use the version from (e.g.)  property-business-api which has better assertions to better pinpoint test failures.